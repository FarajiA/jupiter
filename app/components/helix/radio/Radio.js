import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import { Field } from 'redux-form';

export const Radio = ({ options, fieldName }) => {
  const radioInput = ({ input }) => (
    _.map(options, (option, index) => {
      return (
        <hx-radio-control key={index}>
          <div className="radio-input">
            <input
              {...input}
              id={option.id}
              type="radio"
              value={option.value}
              checked={option.value === input.value}
            />
            <label htmlFor={option.id}>{option.label}</label>
          </div>
        </hx-radio-control>
      );
    }));

  return (
    <div className="radio-options">
      <Field
        component={radioInput}
        name={fieldName}
        options={options}
      />
    </div>
  );
};

Radio.propTypes = {
  fieldName: PropTypes.string.isRequired,
  input: PropTypes.shape({
    value: PropTypes.string
  }),
  options: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      label: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired
    })
  )
};

export default Radio;
