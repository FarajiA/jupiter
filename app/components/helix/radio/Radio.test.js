import { renderWithForm, mountWithForm } from '../../../../test/provider';
import Radio from './Radio';

describe('Radio', () => {
  let wrapper;
  const props = {
    fieldName: 'radio-buttons',
    options: [
      {
        id: 'option-1',
        label: 'First Option',
        value: 'first-option'
      },
      {
        id: 'option-2',
        label: 'Second Option',
        value: 'second-option'
      }
    ]
  };
  beforeEach(() => {
    wrapper = mountWithForm(Radio, { ...props });
  });

  test('renders with appropriate props', () => {
    const component = renderWithForm(Radio, { ...props }).toJSON();
    expect(component).toMatchSnapshot();
  });

  test('it renders a radio for each option', () => {
    expect(wrapper.find('hx-radio-control').length).toEqual(2);
  });

  test('it renders the option labels', () => {
    const labels = wrapper.find('label').map((label) => label.text());
    expect(labels).toEqual([
      'First Option',
      'Second Option'
    ]);
  });

  test('it assigns the value prop correctly', () => {
    const labels = wrapper.find('input').map((label) => label.prop('value'));
    expect(labels).toEqual([
      'first-option',
      'second-option'
    ]);
  });

  test('it assigns the same name for both radio buttons', () => {
    const labels = wrapper.find('input').map((label) => label.prop('name'));
    expect(labels).toEqual([
      'radio-buttons',
      'radio-buttons'
    ]);
  });
});
