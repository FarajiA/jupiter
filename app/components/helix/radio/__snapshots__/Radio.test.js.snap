// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio renders with appropriate props 1`] = `
<div
  className="radio-options"
>
  <hx-radio-control>
    <div
      className="radio-input"
    >
      <input
        checked={false}
        id="option-1"
        name="radio-buttons"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        type="radio"
        value="first-option"
      />
      <label
        htmlFor="option-1"
      >
        First Option
      </label>
    </div>
  </hx-radio-control>
  <hx-radio-control>
    <div
      className="radio-input"
    >
      <input
        checked={false}
        id="option-2"
        name="radio-buttons"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        type="radio"
        value="second-option"
      />
      <label
        htmlFor="option-2"
      >
        Second Option
      </label>
    </div>
  </hx-radio-control>
</div>
`;
